multicluster/cmd/multicluster-controller/clusterset_webhook.go

```diff
@@ -59,11 +59,22 @@ func (v *clusterSetValidator) Handle(ctx context.Context, req admission.Request)
 			klog.ErrorS(err, "Error while decoding old ClusterSet", "ClusterSet", klog.KObj(clusterSet))
 			return admission.Errored(http.StatusBadRequest, err)
 		}
-		if oldClusterSet.Spec.Leaders[0].ClusterID != clusterSet.Spec.Leaders[0].ClusterID {
+
+		if oldClusterSet.Spec.ClusterID != "" && oldClusterSet.Spec.ClusterID != clusterSet.Spec.ClusterID {
 			klog.ErrorS(err, "the field 'clusterID' is immutable", "ClusterSet", klog.KObj(clusterSet))
 			return admission.Denied("the field 'clusterID' is immutable")
 		}
 
+		if oldClusterSet.Spec.ClusterSetID != "" && oldClusterSet.Spec.ClusterSetID != clusterSet.Spec.ClusterSetID {
+			klog.ErrorS(err, "the field 'clusterSetID' is immutable", "ClusterSet", klog.KObj(clusterSet))
+			return admission.Denied("the field 'clusterSetID' is immutable")
+		}
+
+		if oldClusterSet.Spec.Leaders[0].ClusterID != clusterSet.Spec.Leaders[0].ClusterID {
```

Comment:
Would you also check only a single leader?

Reply:
There are already a few marks to limit the number of leader cluster:
```
	// +kubebuilder:validation:MinItems=1
	// +kubebuilder:validation:MaxItems=1
	// +kubebuilder:validation:Required
```